ionic2/app/pages/search/search.page.ts

```diff
@@ -64,7 +64,7 @@ export class SearchPage {
 
   selectPokemon(pokemon:Pokemon) {
     this.cancelRequests();
-    this.navCtrl.setRoot(WikiPage, {pokemonId: pokemon.pokemonId});
+    this.navCtrl.setRoot(PokeDetailPage, {pokemonId: pokemon.pokemonId});
```

Comment:
You actually fixed the wrong one `selectLocation` instead of `selectedPokemon`.
But I also meant something different than what you've implemented. Maybe my explanation wasn't clear, sorry.
What I meant was that when you select a Pokemon from the search results it will modify your navigation stack to look like this
`WikiIndexPage -> PokeDetailPage`
So when you hit the back button it will take you back to the `WikiIndexPage` instead of to the search results. For some reason this seems more intuitive to me. I implemented the behavior myself and pushed the changes. Please tell me what you think about this.